sys/callout.c

```diff
@@ -90,5 +91,12 @@ void callout_process(realtime_t time) {
 
       elem = next;
     }
+    if (current_bucket == last_bucket)
+      break;
+
+    current_bucket = (current_bucket + 1) % CALLOUT_BUCKETS;
   }
+
+  /* Update ci.last. */
```

Comment:
Nonetheless when I had a look at this code I was missing the meaning of `ci.last` – hence I believe adding a comment to `ci` structure would have a positive impact on readability.